research/2022-07-15-relay-anonymity.md

```diff
@@ -0,0 +1,324 @@
+We might cover these in future posts.
+
+### Privacy
+
+Privacy allows users to choose which data and information
+
+* they want to share
+* and with whom they want to share it.
+
+This includes data and information that is associated with and/or generated by users.
+Protected data also comprises metadata that might be generated without users being aware of it.
+
+Often privacy is realized by the confidentiality property of security.
+This neither makes privacy and security the same, nor the one a sub category of the other.
+While security is abstract itself (its properties can be realized in various ways), privacy lives on a more abstract level using security properties.
+Privacy typically does not use integrity and confidentiality. An adversary who has no access to the private data, because the message has been encrypted, could still alter the message.
+
+Waku is privacy-preserving. It offers:
+
+* confidentiality, because message content is only disclosed to the indented receivers.
+  We have various mechanisms in place to guarantee confidentiality.
+  The strongest option are secure channels setup with the help of the [Noise Protocol Framework](https://noiseprotocol.org/).
+* meta-data protection, 
+  - TODO: what meta-data (outside of anonymity related) should be protected by the privacy-preserving property
+
+### Anonymity
+
+Privacy and anonymity are closely linked.
+Both the identity of a user and data that allows inferring a user's identity should be part of the privacy policy.
+For the purpose of analysis, we want to have a clearer separation between these concepts.
+
+We define anonymity as *unlinkablity of users' identities and their shared data and/or actions*.
+
+We subdivide anonymity into *receiver anonymity* and *sender anonymity*.
+
+#### Receiver Anonymity
+
+We define receiver anonymity as *unlinkability of users' identities and the data they receive and/or related actions*.
+The data transmitted via Waku relay must be a [Waku message](https://rfc.vac.dev/spec/14/), which contains a content topic field.
+Because each topic is associated with a content topic, and each receiver is interested in messages with specific content topics,
+receiver anonymity in the context of Waku corresponds to *subscriber-topic unlinkability*.
+An example for the "action" part of our receiver anonymity definition is subscribing to a specific topic.
+
+The Waku message's content topic is not related to the libp2p pubsub topic.
+For now, Waku uses a single libp2p pubsub topic, which means messages are propagated via a single mesh of peers.
+With this, the receiver discloses its participation in Waku on the gossipsub layer.
+We will leave the analysis of libp2p gossipsub to a future article within this series, and only provide a few hints and pointers here.
+
+Waku offers k-anonymity regarding content topic interest in the global adversary model.
```

Comment:
I'd explain the notion of k-anonymity

Also consider introducing anonymity trilemma - make the fundamental trade-off clear

Reply:
https://freedom.cs.purdue.edu/projects/trilemma.html and https://eprint.iacr.org/2017/954.pdf